### DISCUSSION TITLE
Minio memory usage explained

### DISCUSSION THREAD

**Discussion by lukahrobat** [minio/minio#19253] [Q&A] [answered]
To start I am fairly new to Minio so please bear with me.

I am running a Minio instance RELEASE.2024-03-07T00-43-48Z via Podman as a systemd service with a non-root user on Centos Stream 9. The server has 16GB of memory available and 4x2TB disks. for uploads I use Rclone utility.

The memory consumption of Minio container reported by Podman/top spikes to 3-5GB when uploading files and drops to 1-2GB after it's finished which is fine but my total memory usage spikes to 80-90% as reported by the system and remains elevated even after the upload tasks are completed. I have no other container or app running in the background that would consume so much memory. I even tired adding up the memory usage by PID as reported by top but it does not add up to the total usage either.
Interestingly even if I stop the Minio container after uploads the total memory usage doesn't fall either. It just hovers at those elevated levels until I reboot.

I was wondering if Minio is allocating memory which is not displayed on top perhaps? Or is it a Podman related issue?

Thanks in advance!

**Comment by klauspost** [comment, marked as answer]
> I was wondering if Minio is allocating memory which is not displayed on top perhaps? 

No. MinIO runs as a regular process and does nothing magic. 

It is however normal that MinIO keeps buffers after using them and spikes up to 3-5GB is to be expected.

**Reply by lukahrobat** [reply]
If I setup Minio without Podman so natively using the same data directory & config it should not cause any issues or data loss right? I just want to check if I can replicate the behavior without Podman.

**Reply by klauspost** [reply]
Yes. Shouldn't be a problem. Either way it will tell you.
